platform/darwin/src/MGLGeometryOfflineRegion.h

```diff
@@ -0,0 +1,80 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLFoundation.h"
+#import "MGLOfflineRegion.h"
+#import "MGLShape.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ An offline region defined by a style URL, geographic coordinate bounds, and
+ range of zoom levels.
+ */
+MGL_EXPORT
+@interface MGLGeometryOfflineRegion : NSObject <MGLOfflineRegion, NSSecureCoding, NSCopying>
+
+/**
+ URL of the style whose resources are required for offline viewing.
+
+ In addition to the JSON stylesheet, different styles may require different font
+ glyphs, sprite sheets, and other resources.
+
+ The URL may be a full HTTP or HTTPS URL or a Mapbox URL indicating the style’s
+ map ID (`mapbox://styles/{user}/{style}`).
+ */
+@property (nonatomic, readonly) NSURL *styleURL;
+
+/**
+ The geometry bounds for the geographic region covered by the downloaded
+ tiles.
+ */
+@property (nonatomic, readonly) MGLShape *geometry;
```

Comment:
This property would be named `shape` rather than `geometry`.

MGLPointAnnotation is a subclass of MGLShape that corresponds to a Point in GeoJSON. If the region needs to have dimension, we should require that the shape conform to the MGLOverlay protocol (`MGLShape <MGLOverlay> *`), though that would still allow for MGLPolyline. If the region needs to be two-dimensional, then we’d need to introduce either a runtime class check or a new protocol like “MGLSpaceFilling”.